dockstore-client/src/main/java/io/dockstore/client/cli/YamlVerify.java

```diff
@@ -0,0 +1,209 @@
+package io.dockstore.client.cli;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+
+import io.dockstore.client.cli.nested.WorkflowClient;
+import io.dockstore.common.yaml.DockstoreYaml12;
+import io.dockstore.common.yaml.DockstoreYamlHelper;
+import io.dockstore.common.yaml.Service12;
+import io.dockstore.common.yaml.YamlWorkflow;
+import io.swagger.client.ApiException;
+import io.swagger.client.api.UsersApi;
+import io.swagger.client.api.WorkflowsApi;
+import org.yaml.snakeyaml.Yaml;
+
+import static com.google.api.client.util.Objects.equal;
+import static io.dockstore.client.cli.ArgumentUtility.containsHelpRequest;
+import static io.dockstore.client.cli.ArgumentUtility.out;
+import static io.dockstore.client.cli.ArgumentUtility.printFlagHelp;
+import static io.dockstore.client.cli.ArgumentUtility.printHelpFooter;
+import static io.dockstore.client.cli.ArgumentUtility.printHelpHeader;
+import static io.dockstore.client.cli.ArgumentUtility.printLineBreak;
+import static io.dockstore.client.cli.ArgumentUtility.printUsageHelp;
+import static io.dockstore.client.cli.ArgumentUtility.reqVal;
+
+/*
+    GENERAL STEPS:
+    1. Verfiy that YAML syntax is valid
+    2. Verify that all fields are named correctly
+    3. Verify that all required fields are present
+    4. Verify that all files exist
+ */
+
+public class YamlVerify extends WorkflowClient {
```

Comment:
I realize it may be tricky the way the code is structured, and I'm fuzzy on those details, but it doesn't seem like you should be extending WorkflowClient.

1. If I wrote a .dockstore.yml that had only tools and/or services, I would run `dockstore workflow validate`, which seems counterintuitive.
2. In the constructor on line 44, I don't think you use any of those parameters (I could be wrong though).

I'd argue for something like `dockstore validate`, or maybe `dockstore yaml validate`, which could give us the option in the future to do something like `dockstore yaml generate`

Reply:
Good Idea. So I should create a class called `YamlClient` (which would implement ` AbstractEntryClient<T>`) and another class `ValidateYaml` which would extend `YamlClient` and in `ValidateYaml` would be all of the code in `YamlVerify`?

Reply:
> Good Idea. So I should create a class called `YamlClient` (which would implement ` AbstractEntryClient<T>`) and another class `ValidateYaml` which would extend `YamlClient` and in `ValidateYaml` would be all of the code in `YamlVerify`?

I don't think you should extend AbstractEntryClient -- an entry is a tool, workflow, or service, and this feature is sort of global.

It looks like there are a [couple of other clients ](https://github.com/dockstore/dockstore-cli/blob/439478e67c1985c6b8306865251e6e58ac465654/dockstore-client/src/main/java/io/dockstore/client/cli/Client.java#L714)that don't extend AbstractEntryClient; perhaps you could follow that pattern.

Reply:
I have followed a pattern similar to `PluginClient` and `SearchClient`, however `JCommanderUtility` was causing some issue with my use of the require parameter of `--path <path>` so I did it slightly differently.